signer-js/src/index.js

```diff
@@ -0,0 +1,188 @@
+}
+
+function transactionParse(cborMessage, testnet) {
+  // FIXME: Check buffer size and extra bytes
+  // https://github.com/dignifiedquire/borc/issues/47
+  const decoded = cbor.decodeFirst(cborMessage);
+
+  if (decoded[0] !== 0) { throw new Error("Unsupported version") };
+  if (decoded.length < 9) { throw new Error("The cbor is missing some fields... please verify you 9 fields.") };
+
+  let message = {};
+
+  message.to = bytesToAddress(decoded[1], testnet);
+  message.from = bytesToAddress(decoded[2], testnet);
+  message.nonce = decoded[3];
+  message.value = decoded[4].readUIntBE(0,decoded[4].length).toString(10);
+  message.gasprice = decoded[5].readUIntBE(0,decoded[5].length).toString(10);
+  message.gaslimit = decoded[6];
+  message.method = decoded[7];
+  message.params = decoded[8].toString();
+
+  return message;
+}
+
+function transactionSignRaw(unsignedMessage, privateKey) {
+  if (typeof unsignedMessage === 'object') { unsignedMessage = transactionSerializeRaw(unsignedMessage); };
+  if (typeof unsignedMessage === 'string') { unsignedMessage = Buffer.from(unsignedMessage, 'hex'); };
+
+  // verify format and convert to buffer if needed
+  privateKey = tryToPrivateKeyBuffer(privateKey);
+
+  const messageDigest = getDigest(unsignedMessage);
+  const signature = secp256k1.ecdsaSign(messageDigest, privateKey);
+
+  let signatureRSV = Buffer.concat([signature.signature, Buffer.from([signature.recid])]);
+
+  return signatureRSV;
+}
+
+function transactionSign(unsignedMessage, privateKey) {
+  if (typeof unsignedMessage !== 'object') { throw new Error("'message' need to be an object. Cannot be under CBOR format.") };
+  const signature =  transactionSignRaw(unsignedMessage, privateKey);
+
+  let signedMessage = {};
+
+  signedMessage.message = unsignedMessage;
+
+  // FIXME: only support secp256k1
+  signedMessage.signature = {
+    data: signature.toString('base64'),
+    type: 1
+  }
+
+  return signedMessage;
+}
+
+function transactionSignLotus(unsignedMessage, privateKey) {
+  let signedMessage = transactionSign(unsignedMessage, privateKey);
+
+  let signedMessageJSON = JSON.stringify({
+    "Message": {
+      "From": signedMessage.message.from,
+      "GasLimit": signedMessage.message.gaslimit,
+      "GasPrice": signedMessage.message.gasprice,
+      "Method": signedMessage.message.method,
+      "Nonce": signedMessage.message.nonce,
+      "Params": signedMessage.message.params,
+      "To": signedMessage.message.to,
+      "Value": signedMessage.message.value
+    },
+    "Signature": {
+      "Data": signedMessage.signature.data,
+      "Type": signedMessage.signature.type
+    }
+  });
+
+  return signedMessageJSON;
+}
+
+// TODO: new function 'verifySignature(signedMessage)'; Makes more sense ?
+function verifySignature(signature, message) {
+
+  if (typeof message === 'object') { message = transactionSerializeRaw(message); };
+  if (typeof message === 'string') { message = Buffer.from(message, 'hex'); };
+
```

Comment:
If it is something else error?

Reply:
I believe if it is something else `getDigest` function will return an error... but I am not sure.
I am assuming that `hasher.update` only accept Buffer but it might work with a Number too....